cli/ingress-controller/flags.go

```diff
@@ -311,6 +320,9 @@ func parseFlags() (cliConfig, error) {
 	config.LogLevel = viper.GetString("log-level")
 	config.LogFormat = viper.GetString("log-format")
 
+	// Diagnostics
+	config.DumpConfig = util.ParseConfigDumpMode(viper.GetString("dump-config"))
```

Comment:
Neither this flag, nor `util.ParseDumpConfig`, is tested now. I think this deserves a test in `flag_test.go`, involving all possible values of the `dump-config` flag (`"sensitive"`, `"enabled"`, `""`, `"sdifohjwoeui"`)